server/services/v1/row_reader.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/tigrisdata/tigris/internal"
+	"github.com/tigrisdata/tigris/keys"
+	"github.com/tigrisdata/tigris/query/filter"
+	"github.com/tigrisdata/tigris/query/read"
+	qsearch "github.com/tigrisdata/tigris/query/search"
+	"github.com/tigrisdata/tigris/server/transaction"
+	"github.com/tigrisdata/tigris/store/kv"
+	"github.com/tigrisdata/tigris/store/search"
+	ulog "github.com/tigrisdata/tigris/util/log"
+)
+
+type Row struct {
+	Key  []byte
+	Data *internal.TableData
+}
+
+type RowReader interface {
+	NextRow(row *Row) bool
+	Err() error
+}
+
+type SearchRowReader struct {
+	idx    int
+	err    error
+	result *SearchResponse
+}
+
+func MakeSearchRowReader(ctx context.Context, table string, _ []read.Field, filters []filter.Filter, store search.Store) (*SearchRowReader, error) {
+	builder := qsearch.NewBuilder()
+	searchFilter := builder.FromFilter(filters)
+
+	result, err := store.Search(ctx, table, searchFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	var hitsResp = NewHitsResponse()
+	for _, r := range result {
+		hitsResp.Append(r.Hits)
+	}
+
+	var s = &SearchResponse{
+		Hits:   hitsResp,
+		Facets: CreateFacetResponse(result[0].FacetCounts),
+	}
+
+	return &SearchRowReader{
+		idx:    0,
+		result: s,
+	}, nil
+}
+
+func MakeSearchRowReaderUsingFilter(ctx context.Context, table string, filters []filter.Filter, store search.Store) (*SearchRowReader, error) {
+	return MakeSearchRowReader(ctx, table, nil, filters, store)
+}
+
+func (s *SearchRowReader) NextRow(row *Row) bool {
```

Comment:
This method is error prone, we need to refactor it in the future.

Reply:
why?